### PR TITLE
Import csv file for parametric study

### LINKED ISSUE #118: Import csv file for parametric study
### 📝 Description of the feature

Use Case: A user may have a csv file with parametric results that they want to load into a `ParametricStudy` object. Add option to either replace or append data to table. Default is to append.

### 💡 Steps for implementing the feature

_No response_

### 🔗 Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by safeerehman**
closes #118 

- Added a capability to add simulations from a csv file into a parametric study object
- Updated add_inputs to check for duplicate entries
- import_csv_study will add entries if the materials field is the same for all entries in the parametric study object
- added new and updated a few existing unit tests

### CODE REVIEW

File: tests/data/csv/not_a_csv.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+not a csv file
```
Comment (pkrull-ansys) [inline tests/data/csv/not_a_csv.txt:1]: This file is unnecessary. In your test, you can use the `__file__` variable to read the current file, which is not a csv file.

Comment (safeerehman) [reply tests/data/csv/not_a_csv.txt:1]: Yes, works with __file__ as an arg.

File: src/ansys/additive/core/machine.py [hunk 2]
```diff
@@ -141,6 +141,8 @@ def __eq__(self, __o: object) -> bool:
     def __validate_range(self, value, min, max, name):
         if value < min or value > max:
             raise ValueError("{} must be between {} and {}.".format(name, min, max))
+        elif math.isnan(value):
+            raise ValueError("{} must be a number.".format(name))
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/machine.py:145]:
```suggestion
        if math.isnan(value):
            raise ValueError("{} must be a number.".format(name))
        if value < min or value > max:
            raise ValueError("{} must be between {} and {}.".format(name, min, max))
```
This is really nit-picky but I think it's more common to check for `isnan()` prior to checking if a value is within limits.

File: src/ansys/additive/core/microstructure.py [hunk 3]
```diff
@@ -193,6 +193,8 @@ def __eq__(self, __o: object) -> bool:
     def __validate_range(value, min, max, name):
         if value < min or value > max:
             raise ValueError("{} must be between {} and {}.".format(name, min, max))
+        elif math.isnan(value):
+            raise ValueError("{} must be a number.".format(name))
```
Comment (pkrull-ansys) [inline src/ansys/additive/core/microstructure.py:197]:
```suggestion
        if math.isnan(value):
            raise ValueError("{} must be a number.".format(name))
        if value < min or value > max:
            raise ValueError("{} must be between {} and {}.".format(name, min, max))
```

File: src/ansys/additive/core/parametric_study/parametric_study.py [hunk 4]
```diff
@@ -107,6 +108,31 @@ def _new(cls, study_path: pathlib.Path):
         study._init_new_study(study_path)
         return study
 
+    def import_csv_study(self, file_name: str | os.PathLike):
+        """Import a csv file into the parametric study.
+
+        file_name: str, os.PathLike
+             Name of the csv file the simulation parameters are written to.
```
[the remark this answers is not in the document]
Comment (pkrull-ansys) [reply src/ansys/additive/core/parametric_study/parametric_study.py:115]: For this doc string, let's add a link to  `ColumnNames` in the parametric_study/constants.py so folks know what columns are expected.

**Review by pkrull-ansys — CHANGES_REQUESTED** [review]

**Comment by safeerehman** [comment]
Closing this PR as work was covered in PR #250

**Pull request closed without merging**